Automattic/portkey-android
# Stories library part 7 - Error notification intent Loader interface

This PR builds on top of https://github.com/Automattic/portkey-android/pull/352

Related WPAndroid PR https://github.com/wordpress-mobile/WordPress-Android/pull/11987

This PR implements a new interface called `NotificationIntentLoader` so that the Activity implementing `ComposeLoopFrameActivity` (or other entities in the host app) is provided in the intent for error notifications.
When you tap on an error notification, the intent should launch the Composer activity, but this can only be brought by knowing both the name of the class implementing / extending the abstract class `ComposeLoopFrameActivity`, and with that it can also  be passed any extras the host app may need.
Hence, the best solution is to ask for the host app to provide such intent with all its extras and flags set.
You can see how this is done in the demo app, but its use is clearer when done in the context of the WPAndroid app (see the related WPANdroid PR https://github.com/wordpress-mobile/WordPress-Android/pull/11987).

To test:

0. introduce some error in saving a frame, artificially for example [use this gist ](https://gist.github.com/mzorz/80541e6f095ba9785ce0ab0ce1baf4df) to make image saving fail
1. open the demo app
2. add an image (either with camera or uploading)
3. add some text
4. tap on PUBLISH
5. when the error notification appears, tap on it
6. observe the error handling screen appears
